Dockerfile

```diff
@@ -1,62 +1,56 @@
-FROM python:3.10-slim
+FROM nvidia/cuda:12.0.1-base-ubuntu22.04
 
 # Set environment variables
-ENV PYTHONDONTWRITEBYTECODE=1 \
-    PYTHONUNBUFFERED=1 \
-    DEBIAN_FRONTEND=noninteractive \
-    PIP_NO_CACHE_DIR=1 \
-    CONFIG_PATH=/config/config.json
-
-# Create app user
-RUN groupadd -r squishy && \
-    useradd -r -g squishy -d /app -s /bin/bash squishy
-
-# Install system dependencies
-RUN apt-get update && \
-    apt-get install -y --no-install-recommends \
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PYTHONUNBUFFERED=1
+ENV PYTHONDONTWRITEBYTECODE=1
+
+# Install system dependencies including ffmpeg and CUDA development libraries
+RUN apt-get update && apt-get install -y \
+    python3 \
+    python3-pip \
+    python3-dev \
     ffmpeg \
-    libva-dev \
-    va-driver-all \
-    mesa-va-drivers \
-    intel-media-va-driver \
-    i965-va-driver \
-    libva-drm2 \
-    libva-x11-2 \
-    libdrm2 \
-    libdrm-intel1 \
-    libvdpau1 \
-    ocl-icd-opencl-dev \
-    vainfo \
-    && apt-get clean \
+    libavcodec-dev \
+    libavformat-dev \
+    libavutil-dev \
+    libswscale-dev \
+    nvidia-cuda-toolkit \
+    nvidia-cuda-dev \
+    libavfilter-dev \
     && rm -rf /var/lib/apt/lists/*
 
-# Create necessary directories
-RUN mkdir -p /app && \
-    chown -R squishy:squishy /app
-
-# Set working directory
+# Create app directory
 WORKDIR /app
 
-# Copy project files
+# Copy requirements and install
+COPY pyproject.toml /app/
+RUN pip3 install --no-cache-dir -e .
+
+# Copy application code
 COPY . /app/
-RUN chown -R squishy:squishy /app
 
-# Copy entrypoint script
+# Create directories if they don't exist
+RUN mkdir -p /media /transcode
```

Comment:
Do not assume paths for media and transcoding output. The user has an opportunity to define these during the onboarding flow for the app.